lib/autoupdate.ps1

```diff
@@ -431,3 +424,104 @@ function autoupdate([String] $app, $dir, $json, [String] $version, [Hashtable] $
         Write-Host -f DarkGray "No updates for $app"
     }
 }
+
+## Helper Functions
+
+function PropertyHelper {
+    <#
+    .SYNOPSIS
+        Helper of updating property
+    .DESCRIPTION
+        Update manifest property (String, Array or PSCustomObject).
+    .PARAMETER Property
+        Property to be updated
+    .PARAMETER Value
+        New property values
+        Update line by line
+    .OUTPUTS
+        System.Object[]
+            The first element is new property, the second element is change flag
+    #>
+    param (
+        [Object]$Property,
+        [Object]$Value
+    )
+    $hasChanged = $false
+    switch ($Property.GetType().Name) {
+        'String' {
+            $Value = $Value -as [String]
```

Comment:
Is this cast needed?

Reply:
This is for error handling, i.e., if `$Property` is `[String]` but `$Value` is `[Object]` in mistake, then `$Value` will be `$null` and just do nothing.

Reply:
This cause the following `extract_dir` bug, if `$Value` is `[Array]` and `$Property` is `[String]`. I'll check if `$Value = $Value[0]` works.

Reply:
Fixed with length comparison.